awx_collection/plugins/modules/tower_credential.py

```diff
@@ -257,162 +274,117 @@
-                        module.params['ssh_key_data'] = to_text(f.read())
-                else:
-                    module.params['ssh_key_data'] = data
-
-            if module.params.get('vault_id', None) and module.params.get('kind') != 'vault':
-                module.fail_json(msg="Parameter 'vault_id' is only valid if parameter 'kind' is specified as 'vault'")
-
-            for key in OLD_INPUT_NAMES:
-                if 'kind' in params:
-                    params[key] = module.params.get(key)
-                elif module.params.get(key):
-                    params.setdefault('inputs', {})[key] = module.params.get(key)
-
-            if state == 'present':
-                result = credential.modify(**params)
-                json_output['id'] = result['id']
-            elif state == 'absent':
-                result = credential.delete(**params)
-        except (exc.NotFound) as excinfo:
-            module.fail_json(msg='Failed to update credential, organization not found: {0}'.format(excinfo), changed=False)
-        except (exc.ConnectionError, exc.BadRequest, exc.AuthError) as excinfo:
-            module.fail_json(msg='Failed to update credential: {0}'.format(excinfo), changed=False)
-
-    json_output['changed'] = result['changed']
-    module.exit_json(**json_output)
+    # Attempt to look up the related items the user specified (these will fail the module if not found)
+    if organization:
+        org_id = module.resolve_name_to_id('organizations', organization)
+    if user:
+        user_id = module.resolve_name_to_id('users', user)
+    if team:
+        team_id = module.resolve_name_to_id('teams', team)
+
+    if kind:
+        module.deprecate(msg='The kind parameter has been depricated, please use credential_type instead', version="3.6")
+
+    cred_type_id = module.resolve_name_to_id('credential_types', credential_type if credential_type else KIND_CHOICES[kind])
+
+    # Attempt to look up the object based on the provided name and inventory ID
+    credential = module.get_one('credentials', **{
+        'data': {
+            'name': name,
+            'credential_type': cred_type_id,
+        }
+    })
+
+    # Create credential input from legacy inputs
+    credential_inputs = {}
+    for legacy_input in OLD_INPUT_NAMES:
+        if module.params.get(legacy_input) is not None:
+            module.deprecate(msg='{0} parameter has been depricated, please use inputs instead'.format(legacy_input), version="3.6")
+            credential_inputs[legacy_input] = module.params.get(legacy_input)
+    if inputs:
+        credential_inputs.update(inputs)
+
+    # Create the data that gets sent for create and update
+    credential_fields = {
+        'name': new_name if new_name else name,
+        'credential_type': cred_type_id,
+        'inputs': credential_inputs,
+    }
+    if description:
+        credential_fields['description'] = description
+    if organization:
+        credential_fields['organization'] = org_id
```

Comment:
IIRC this can't be changed, but if it's added to the lookup fields that should become a non-issue.

Reply:
Added to the lookup fields.